_layouts/benchmarks.html

```diff
@@ -0,0 +1,78 @@
+---
+layout: default
+---
+
+<section>
+    <div class="w-80 m-auto mt-2">
+        <div class="d-md-flex">
+            <div class="d-sm-none">
+                <img class="logo" src="{{site.baseurl}}/assets/img/rust-python-logo.svg" alt="RustPython Logo">
+            </div>
+            <div class="pl-md-2">
+                <div class="section-title">RustPython</div>
+                <div class="title">{{ page.title | escape }}</div>
+                <small>{{ site.benchmarks-intro }}</small>
+            </div>
+        </div>
+    </div>
+</section>
+
+<div class="w-80 m-auto mt-2">
+    <div class="benchmarks-intro">
+        {{ content }}
+    </div>
+
+    {% assign folders = "" | split: ", " %}
+    {% for image in site.static_files %}
+        {% if image.path contains 'criterion/' %}
+            {% assign imagepath = image.path | split: "/" %}
+            {% for subpath in imagepath %}
+                {% if forloop.index0 == 3 %}
+                    {% assign folders = folders | append: pathName %}
+                {% endif %}
+            {% endfor %}
+        {% endif %}
+    {% endfor %}
+    {% assign folders = folders | uniq %}
+
+    <!-- The code block below does exactly the same as the one above, which one is cleaner? -->
+    <!-- {% assign folders = "" | split: ", " %}
+    {% for folder in site.static_files %}
+    {% if folder.path contains 'criterion/' %}
+    {% assign temp = folder.path | split: 'criterion/' %}
+    {% assign pathName = temp[1] | split: '/' | first %}
+    {% assign pathName = pathName | split: ", " %}
+    {% assign folders = folders | concat: pathName %}
+    {% endif %}
+    {% endfor%}
+    {% assign folders = folders | uniq %} -->

```

Comment:
The commented code loops through the folders and looks for a "criterion" directory.

The first code block loops through all the images and checks if any of them has `criterion` in it as a folder name. I don't see any break for the loop if something is found.

IMHO the commented code is more efficient, because there might be tons images, but def. less directories... so the commented code actually loops less.

This is about the logic of which approach is better.... i am not sure if others agree.

Reply:
I agree, thanks for checking it out. I've deleted the less efficient code block and uncommented the 'better' one.